pegen/c_generator.py

```diff
@@ -145,6 +146,31 @@ class CParserGenerator(ParserGenerator, GrammarVisitor):
     def __init__(self, rules: Dict[str, grammar.Rule], file: Optional[IO[Text]]):
         super().__init__(rules, file)
         self.callmakervisitor = CCallMakerVisitor(self)
+        self._varname_counter = itertools.count()
+
+    def unique_varname(self, name="tmpvar"):
+        return name + "_" + str(next(self._varname_counter))
+
+    def call_with_errorcheck(self, call_text, returnval="", goto_target=None):
+        error_var = self.unique_varname()
+        self.print(f"int {error_var} = {call_text}")
+        self.print(f"if ({error_var}) {{")
+        with self.indent():
```

Comment:
Not super fan of this way of adding error handling, I want to think about if there is a better abstraction. Suggestions welcome :)